### PR TITLE
azuread_application, azuread_service_principal: docs note to clarify tags usage

### LINKED ISSUE #986: Tags not working for azuread_service_principal
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritise the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureAD Provider) Version

2.33.0

### Affected Resource(s)

azuread_service_principal

### Terraform Configuration Files
```
     azuread = {
      source = "hashicorp/azuread"
      version = "2.33.0"
    }
 

resource "azuread_service_principal" "asp" {
  application_id               = azuread_application.aa.application_id
  app_role_assignment_required = false
  use_existing                 = true
  tags = [
    "CreatedBy_Terraform",
    "AssetID_${var.asset_id}"
  ]
  
}
```

### Debug Output
```
      + tags                         = [
          + "AssetID_1234",
          + "CreatedBy_Terraform"
        ]
      + type                         = (known after apply)
      + use_existing                 = true

      + feature_tags {
          + custom_single_sign_on = (known after apply)
          + enterprise            = (known after apply)
          + gallery               = (known after apply)
          + hide                  = (known after apply)
        }
```
### Expected Behavior

Tags strings should be created on the resource.

### Actual Behavior

No strings are listed in the tags section for the resource. 

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Why is the feature_tags shown in the output when they are not ref in the code

### References

* #0000

### PR DESCRIPTION
**Pull request by manicminer**
Closes #986

### CODE REVIEW

File: docs/resources/application.md [hunk 1]
```diff
@@ -191,7 +191,7 @@ The following arguments are supported:
 
 * `single_page_application` - (Optional) A `single_page_application` block as documented below, which configures single-page application (SPA) related settings for this application.
 * `support_url` - (Optional) URL of the application's support page.
-* `tags` - (Optional) A set of tags to apply to the application. Cannot be used together with the `feature_tags` block.
+* `tags` - (Optional) A set of tags to apply to the application, for configuring specific behaviors of the application and linked service principals. Note that these are not provided for use by practitioners. Cannot be used together with the `feature_tags` block.
```
Comment (jackofallops) [inline docs/resources/application.md:194]: I think the coma is unnecessary here?
```suggestion
* `tags` - (Optional) A set of tags to apply to the application for configuring specific behaviours of the application and linked service principals. Note that these are not provided for use by practitioners. Cannot be used together with the `feature_tags` block.
```

File: docs/resources/service_principal.md [hunk 2]
```diff
@@ -108,7 +108,7 @@ The following arguments are supported:
 
 * `preferred_single_sign_on_mode` - (Optional) The single sign-on mode configured for this application. Azure AD uses the preferred single sign-on mode to launch the application from Microsoft 365 or the Azure AD My Apps. Supported values are `oidc`, `password`, `saml` or `notSupported`. Omit this property or specify a blank string to unset.
 * `saml_single_sign_on` - (Optional) A `saml_single_sign_on` block as documented below.
-* `tags` - (Optional) A set of tags to apply to the service principal. Cannot be used together with the `feature_tags` block.
+* `tags` - (Optional) A set of tags to apply to the service principal, for configuring specific behaviors of the service principal. Note that these are not provided for use by practitioners. Cannot be used together with the `feature_tags` block.
```
Comment (jackofallops) [inline docs/resources/service_principal.md:111]: Same here.
```suggestion
* `tags` - (Optional) A set of tags to apply to the service principal for configuring specific behaviours of the service principal. Note that these are not provided for use by practitioners. Cannot be used together with the `feature_tags` block.
```

**Review by jackofallops — APPROVED** [review]
Couple of observations below, otherwise LGTM 👍

File: docs/resources/application.md (hunk 1 above)
Comment (manicminer) [reply docs/resources/application.md:194]: Good shout, thanks!

**Pull request merged**